db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/SimpleLocaleStringOperatorNode.java

```diff
@@ -0,0 +1,164 @@
+	 * @param rightOperand		The locale
+	 * @param methodName	The method name
+	 */
+
+	public void init(Object leftOperand, Object rightOperand, Object methodName)
+	{
+		super.init(leftOperand, rightOperand, "upperWithLocale", methodName, ClassName.StringDataValue,
+				ClassName.StringDataValue);
+	}
+
+	/**
+	 * Bind this operator
+	 *
+	 * @param fromList			The query's FROM list
+	 * @param subqueryList		The subquery list being built as we find SubqueryNodes
+	 * @param aggregateVector	The aggregate vector being built as we find AggregateNodes
+	 *
+	 * @return	The new top of the expression tree.
+	 *
+	 * @exception StandardException		Thrown on error
+	 */
+	@Override
+	public ValueNode bindExpression(FromList fromList,
+									SubqueryList subqueryList,
+									List<AggregateNode> aggregateVector) throws StandardException {
+		leftOperand = leftOperand.bindExpression(fromList, subqueryList,
+				aggregateVector);
+		rightOperand = rightOperand.bindExpression(fromList, subqueryList,
+				aggregateVector);
+
+		if( leftOperand.requiresTypeFromContext())
+		{
+			leftOperand.setType(new DataTypeDescriptor(TypeId.getBuiltInTypeId(Types.VARCHAR), true));
+			leftOperand.setCollationUsingCompilationSchema();
+		}
+
+		if( rightOperand.requiresTypeFromContext())
+		{
+			rightOperand.setType(new DataTypeDescriptor(TypeId.getBuiltInTypeId(Types.VARCHAR), false));
+			rightOperand.setCollationUsingCompilationSchema();
+		}
+
+
+		/*
+		** Check the type of the operand - this function is allowed only on
+		** string value (char and bit) types.
+		*/
+		TypeId operandType = leftOperand.getTypeId();
+
+		switch (operandType.getJDBCTypeId())
+		{
+				case Types.CHAR:
+				case Types.VARCHAR:
+				case Types.LONGVARCHAR:
+				case Types.CLOB:
+					break;
+				case Types.JAVA_OBJECT:
+				case Types.OTHER:	
+				{
+					throw StandardException.newException(SQLState.LANG_UNARY_FUNCTION_BAD_TYPE, 
+										methodName,
+										operandType.getSQLTypeName());
+				}
+
+				default:
+					DataTypeDescriptor dtd = DataTypeDescriptor.getBuiltInDataTypeDescriptor(Types.VARCHAR, true, 
+							  leftOperand.getTypeCompiler().
+								getCastToCharWidth(
+									leftOperand.getTypeServices()));
+			
+					leftOperand =  (ValueNode)
+						getNodeFactory().getNode(
+							C_NodeTypes.CAST_NODE,
+							leftOperand,
+							dtd,
+							getContextManager());
+					
+				// DERBY-2910 - Match current schema collation for implicit cast as we do for
+				// explicit casts per SQL Spec 6.12 (10)					
+			    leftOperand.setCollationUsingCompilationSchema();
+			    
+				((CastNode) leftOperand).bindCastNodeOnly();
+					operandType = leftOperand.getTypeId();
+		}
+
+		/*
+		** The result type of upper()/lower() is the type of the operand.
+		*/
+
+		setType(new DataTypeDescriptor(operandType,
+				leftOperand.getTypeServices().isNullable(),
+				leftOperand.getTypeCompiler().
+					getCastToCharWidth(leftOperand.getTypeServices())
+						)
+				);
```

Comment:
This code is the reason the following SQL doesn't display properly:
`             values upper('ß', 'de-DE');`
Since the String result isn't created as unicode, it converts to two uppercase S's.  So, the result string could be up to twice as long as the source.  So we need to do something like the following instead:

          TypeId newTypeId = null;
		if (leftOperand.getTypeCompiler().
			getCastToCharWidth(leftOperand.getTypeServices()) * 2 > Limits.DB2_LOB_MAXWIDTH) {
			// throw new StandardException();
		}
		int maxWidth = leftOperand.getTypeCompiler().
			           getCastToCharWidth(leftOperand.getTypeServices()) * 2;

		if (maxWidth > Limits.DB2_VARCHAR_MAXWIDTH) {
			newTypeId = TypeId.getBuiltInTypeId(TypeId.CLOB_NAME);
		}
		else {
			newTypeId = TypeId.getBuiltInTypeId(TypeId.VARCHAR_NAME);
		}

		setType(new DataTypeDescriptor(newTypeId, true, maxWidth));